apps/balancer-tools/src/contexts/PoolSimulatorContext.tsx

```diff
@@ -159,9 +158,9 @@ export function PoolSimulatorProvider({ children }: PropsWithChildren) {
     setData(importedData);
   }
 
-  // useEffect(() => {
-  // if (pathname === "/poolsimulator/analysis") push(generateURL());
-  // }, [pathname]);
+  useEffect(() => {
+    if (isAnalysis) push(generateURL());
+  }, [customData, initialData]);
```

Comment:
🤔 are we always adding this to the URL? Doesn't it impact performance rn? If it does I'd suggest we stop doing this and instead only leave the Share button working